R/Renderer.R

```diff
@@ -23,17 +23,23 @@ Renderer <- R6::R6Class( # nolint: object_name_linter.
     #'
     #' @param blocks `list` of `c("TextBlock", "PictureBlock", "NewpageBlock")` objects.
     #' @param yaml_header `character` a `rmarkdown` `yaml` header.
-    #' @param global_knitr `list` a global `knitr` parameters, like echo.
-    #' But if local parameter is set it will have priority.
-    #' Defaults to empty `list()`.
+    #' @param global_knitr `list` a global `knitr` parameters for customizing the rendering process.
     #'
     #' @return `character` a `Rmd` text (`yaml` header + body), ready to be rendered.
-    renderRmd = function(blocks, yaml_header, global_knitr = list()) {
+    renderRmd = function(blocks, yaml_header, global_knitr = getOption("teal.reporter.global_knitr")) {
       checkmate::assert_list(blocks, c("TextBlock", "PictureBlock", "NewpageBlock", "TableBlock", "RcodeBlock"))
+      checkmate::assert_subset(names(global_knitr), names(knitr::opts_chunk$get()))
+
       if (missing(yaml_header)) {
         yaml_header <- md_header(yaml::as.yaml(list(title = "Report")))
       }
 
+      if (requireNamespace("formatR", quietly = TRUE)) {
+        global_knitr[["tidy"]] <- TRUE
+      } else {
+        message("For better code formatting, consider installing the formatR package.")
```

Comment:
does that mean that I will have a message on every render of report (or even worse: card)?

can we remove it?

Reply:
This message will be displayed in the console for every report rendering, except for the card. While we can remove it from the card, I believe it's important to inform the developer . IF we don't want to inform for every rendering then I can move the message to the .onLoad() function, ensuring it's displayed only once per user session.

Reply:
I see your point. I still think we should notify the app-developer as we are using formatR functionality However, I agree it shouldn't appear during rendering. Perhaps, during the .onload() event unless there are  reasons not to do so.